### ISSUE TITLE
Add a hasZeroData selector for AdSense

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#4674]
## Feature Description

For the new Zero States features to be properly implemented, we need a way to explicitly check if a module has "zero data" as opposed to "gathering data". Obviously any site that's gathering data will also have "zero data", but we can use logic tailored to each "module" (eg Google Service) that Site Kit uses to check if there's no data available from the Google API because the site is new or if there's data because there's been nothing of note.

Essentially, we want a selector for AdSense that allows us to easily know: "Is this user's (default) report lacking any data and returning all zeros?"

I don't _think_ we need an `isGatheringData()` selector for AdSense here, because if AdSense is still gathering data about your site to create an account, that will be a specific AdSense account state and we won't show widgets/fetch AdSense reports yet. But having a convenient selector for empty reports would still be useful.

Upon reviewing the [Figma Designs](https://www.figma.com/file/lPdnmgVoPqOEO82AMle9gH/Zero-State?node-id=435%3A1677) it's actually possible we won't need this at all, so I'm leaving it in Triage for now.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should be a `hasZeroData()` selector for the AdSense module's datastore that checks to see if there is no data for AdSense available in a representative report.
* This selector should `return false` if any data is found. The selector should `return true` if no data for a "representative" AdSense report request—likely whatever the default dashboard request would be (eg. 28 days)—is found but `isGatheringData() === false`.

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by jimmymadon** [comment]
@felixarntz Just wanted to check if this issue is still needed? As @tofumatt has mentioned in the description, it appears that the only place where this selector would be used is [going to be removed from the Figma designs](https://www.figma.com/file/lPdnmgVoPqOEO82AMle9gH/Zero-State?node-id=435%3A1677) (left widget under Earnings).

**Comment by felixarntz** [comment]
@jimmymadon Good point. This also relates to #4087 which we've kept in the backlog for now as well. It's indeed not critical now, so let's hold off this one for now, I'll also remove it from the zero states epic for that reason.

I'll also mark it blocked by #4087.

cc @tofumatt
